source/administration/identity-access-management/oidc-access-management.rst

```diff
@@ -15,18 +15,64 @@ MinIO supports using an OpenID Connect (OIDC) compatible IDentity Provider (IDP)
 Users managed by an OIDC provider must specify the necessary policies as part of the JWT claim. If the user JWT claim has no matching MinIO policies, that user has no permissions to access any action or resource on the MinIO deployment.
 
 The specific claim which MinIO looks for is configured as part of :ref:`deploying the cluster with OIDC identity management <minio-external-iam-oidc>`. This page focuses on creating MinIO policies to match the configured OIDC claims.
 
-Authentication and Authorization Flow
+Authentication and Authorization Flow 
 -------------------------------------
 
+A MinIO cluster can have any number of Role Policy based OpenID providers configured.
+
+A MinIO Cluster may only have **one** JWT Claim based OpenID provider configured.
+Other providers must be configured with the Role Policy method instead.
+
+Role Policy and RoleARN
+~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO recommends using the Role Policy method for authenticating with an OpenID provider.
+With a Role Policy, all registered users receive the same policy or policies for access.
+
+Use :ref:`OpenID Policy Variables <minio-policy-variables-oidc>` to manage what each individual user has access to.
+
+The login flow for an application using :abbr:`OIDC (OpenID Connect)` credentials with a Role Policy claim flow is as follows:
+
+1. Authenticate to the configured :abbr:`OIDC (OpenID Connect)` provider and redirect back to MinIO. 
+   
+   MinIO only supports the `OpenID Authorization Code Flow <https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth>`__. 
+   Authentication using Implicit Flow is not supported.
+
+2. Specify the ``RoleARN`` to the MinIO Security Token Service (STS) :ref:`minio-sts-assumerolewithwebidentity` API endpoint. 
+   
+   .. note::
+
+      MinIO assigns and displays a unique RoleARN to each configured OpenID provider at startup.
+
+   MinIO verifies the ``RoleARN`` and checks for a :ref:`role policy <minio-external-identity-management-openid-access-control>` specifying a list
+   of one or more :ref:`policies <minio-policy>` to assign to the authenticated user. 
+
+   Use either the :envvar:`MINIO_IDENTITY_OPENID_ROLE_POLICY` environment variable or the :mc-conf:`identity_openid role_policy <identity_openid.role_policy>` configuration setting to define the list of policies to use for the provider
+
+3. MinIO returns temporary credentials in the STS API response in the form of an access key, secret key, and session token. 
+   The credentials have permissions matching those policies specified in the Role Policy.
+   
+4. Applications use the temporary credentials returned by the STS endpoint to perform authenticated S3 operations on MinIO.
+
+
+JSON Web Token Claim
+~~~~~~~~~~~~~~~~~~~~
+
+Using JSON Web Tokens allows for individual assignment of policies at the increased cost of managing multiple policies for separate claims.
```

Comment:
This feels like it wants a comma?
```suggestion
Using JSON Web Tokens allows for individual assignment of policies, at the increased cost of managing multiple policies for separate claims.
```

Reply:
This actually doesn't call for a comma. But let me see if I can break it up some.